libs/base/src/lib/context/AccountProvider.tsx

```diff
@@ -249,8 +256,43 @@ const OnboardProvider: FC<{
         console.error(error)
       })
     }
+
+    fetch(`${API_ENDPOINT}/signature`)
+      .then(resp => resp.json())
+      .then(json => {
+        setStakeSignatures(prevSignatures => ({
+          ...prevSignatures,
+          message: json.message,
+        }))
+      })
+      .catch(console.error)
   })
 
+  useEffect(() => {
+    const fetchSignature = async () => {
+      if (!address) return
+
+      fetch(`${API_ENDPOINT}/signature/${address}`)
+        .then(resp => resp.json())
+        .then(json => {
+          if (json.error) return
+          setStakeSignatures(prevSignatures => {
+            // TODO: I'm getting a weird race condition here with the library, this fix the issue
```

Comment:
@JamesLefrere @chrisjgf any better idea on how to deal with this? :)

Reply:
Hmm, I don't see why you would need to put it into an object first (re `prevHack`). Console log the obj blobs & compare?